llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8958,6 +8968,9 @@ static void addScalarResumePhis(VPRecipeBuilder &Builder, VPlan &Plan) {
     // start value provides the value if the loop is bypassed.
     bool IsFOR = isa<VPFirstOrderRecurrencePHIRecipe>(VectorPhiR);
     auto *ResumeFromVectorLoop = VectorPhiR->getBackedgeValue();
+    assert(!Plan.getEarlyExit() &&
+           "Cannot handle reductions or first-order recurrences with "
+           "uncountable early exits");
```

Comment:
This could be added separately, as NFC?

Reply:
Sure, will do although I'll have to check for early exit differently.

Reply:
Created https://github.com/llvm/llvm-project/pull/122732. Will update this PR soon!